### PR TITLE
[1/N] Apply clang-tidy to aten/src/ATen/core/

### PR DESCRIPTION
**Pull request by cyyever**
It is time to clang-tidy aten.

### CODE REVIEW

File: aten/src/ATen/core/dynamic_type.cpp [hunk 1]
```diff
@@ -365,7 +365,7 @@ DynamicType::Ptr IValue::TagType<c10::DynamicType>::get(const c10::IValue& v) {
 
 DynamicTypePtr ivalue::TupleTypeFactory<c10::DynamicType>::create(
     std::vector<TypePtr> elemTypes) {
-  return DynamicTypeFactory::create<TupleType>(std::move(elemTypes));
+  return DynamicTypeFactory::create<TupleType>(elemTypes);
```
Comment (Skylion007) [inline aten/src/ATen/core/dynamic_type.cpp:368]: Hmm, create<> doesn't have a ctor? Interesting. TypePtr is not trivially copyable so shouldn't this be const ref?

**Review by Skylion007 — APPROVED** [review]

**Comment by Skylion007** [comment]
Not ready to enable the clang-tidy lintrunner on it yet?

**Comment by cyyever** [comment]
> Not ready to enable the clang-tidy lintrunner on it yet?

May be in the next PR

File: aten/src/ATen/core/dynamic_type.cpp (hunk 1 above)
Comment (cyyever) [reply aten/src/ATen/core/dynamic_type.cpp:368]: Will check in the next patch

**Pull request closed without merging**